go/pkg/bertyprotocol/scenario_test.go

```diff
@@ -779,6 +782,64 @@ func addAsContact(ctx context.Context, t *testing.T, senders, receivers []*Testi
 	logTree(t, "duration: %s", 0, false, time.Since(start))
 }
 
+func WaitingForContactReady(ctx context.Context, t *testing.T, senders, receivers []*TestingProtocol) {
+	logTree(t, "Waiting that Senders send secrets to Receivers", 0, true)
+	start := time.Now()
+	var waitingSecretsDuration time.Duration
+
+	for _, sender := range senders {
+		for _, receiver := range receivers {
```

Comment:
not sure if this is a slow or not, but if yes, maybe you can use https://pkg.go.dev/golang.org/x/sync/errgroup